testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6315,6 +6315,30 @@ void assertSubFunReturns(boolean binary, String s, int start,
     f12.checkNull("nvl(CAST(NULL AS VARCHAR(6)), cast(NULL AS VARCHAR(4)))");
   }
 
+
+  @Test void testIfNullCoalesceFunc() {
+    final SqlOperatorFixture f = fixture()
+        .withLibrary(SqlLibrary.BIG_QUERY)
+        .setFor(SqlLibraryOperators.IFNULL, VM_EXPAND);
+
+    f.checkString("ifnull('a','b')", "a", "CHAR(1) NOT NULL");
+    f.checkString("ifnull(null,'b')", "b", "CHAR(1)");
```

Comment:
nit: I'm a bit surprised that this isn't `NOT NULL`, since the javadoc for `LEAST_NULLABLE` says:

"the result is NOT NULL if any of the arguments is NOT NULL; otherwise the type is unchanged"

and it would seem the type of `'b'` should be `NOT NULL`. Any ideas as to why? Not sure if this is really a problem or not.

Reply:
Updated to follow NVL pattern and the result is CHAR(1) NOT NULL, so fixed!